### PR TITLE
Closes #40. Enhancement: Run Composer optimize autoloader after DI compile.

### LINKED ISSUE #40: Enhancement: Run Composer optimize autoloader after DI compile
I haven't tested it, but apparently `composer dumpautoload -o --apcu`can be run after DI compile to speed up PHP class lookups. Might be worth looking into.

> "File system is a bottleneck - make sure composer is optimized and class map is generated after setup di compile. Refer to this tweet of mine:"

https://twitter.com/IvanChepurnyi/status/1179280000400465920

### PR DESCRIPTION
**Pull request by erfanimani**
I'm merging into master since develop seems behind.

Disclaimer — I haven't run this in production yet, so give me a few days and I'll can confirm on my end if this works.

### CODE REVIEW

File: recipe/magento_2_1/files.php [hunk 1]
```diff
@@ -14,6 +14,7 @@
 set('static_deploy_options', '--exclude-theme=Magento/blank');
 
 task('files:compile', '{{bin/php}} {{magento_bin}} setup:di:compile');
+task('files:optimize-autoloader', '{{bin/composer}} dump-autoload --optimize --apcu');
 task('files:static_assets', '{{bin/php}} {{magento_bin}} setup:static-content:deploy {{languages}} {{static_deploy_options}}');
 task(
     'files:permissions',
```
Comment (erfanimani) [inline recipe/magento_2_1/files.php]: Is the cd into `magento_dir` necessary? Other commands don't do this.

**Comment by sprankhub** [comment]
Awesome, thanks!

I am always confused regarding the file structure in this repo. So... Does this also work for 2.2 and 2.3 deployments?

**Comment by erfanimani** [comment]
Haha same here. Yeah, 2.1 is inherited by both 2.2 and 2.3. (Although I probably won't test it with 2.1 or 2.2, it should work with previous versions of Magento)

**Comment by erfanimani** [comment]
I made some changes, and it's working properly now. Tested on two projects.

To test, after a deployment open up vendor/composer/autoload_classmap.php and ensure classes exist that are located in `/generated/code` (should be a couple thousand occurrences)

**Comment by osrecio** [comment]
Thanks @erfanimani I will merge as soon as possible.

I would also like to take the opportunity to remove the support for Magento 2.0 and 2.1 as Magento itself does not support them.

Taking advantage of that. I will think about how to simplify the repository structure to make it clearer.

**Comment by erfanimani** [comment]
Thanks @osrecio , much appreciated

**Comment by osrecio** [comment]
Your PR  is merged @erfanimani 

Release 2.7.0

**Pull request merged**